File: src/Application.Operation/Features/Admin/ProfileLogs/DTOs/ProfileLogDto.cs
using Tawtheef.Domain.Entities.Recruitment;

namespace Application.Operation.Features.Admin.ProfileLogs.DTOs;

public sealed record ProfileLogDto
{
    public Guid Id { get; init; }
    public Guid UserProfileId { get; init; }
    public required string Source { get; init; }
    public required string ActionType { get; init; }
    public string? Section { get; init; }
    public string? Notes { get; init; }
    public Guid? UserId { get; init; }
    public string? UserName { get; init; }
    public Guid? EntityId { get; init; }
    public Guid? AttachmentId { get; init; }
    public ReviewStatus? ReviewStatus { get; init; }
    public DateTimeOffset CreatedDate { get; init; }
}

File: src/Tawtheef.Websites/operations-web/src/app/pages/user/employee/profile-logs/models/profile-log.dto.ts
import { ReviewStatus } from '../../profile-managment/approval-list/models/profile-approval.models';

export interface ProfileLogDto {
  id: string;
  userProfileId: string;
  source: string;
  actionType: string;
  section?: string | null;
  notes?: string | null;
  userId?: string | null;
  userName?: string | null;
  userProfileOwnerName?: string | null;
  entityId?: string | null;
  attachmentId?: string | null;
  reviewStatus?: ReviewStatus | null;
  createdDate: string;
}
